src/librustc/ty/error.rs

```diff
@@ -144,8 +144,7 @@ impl<'tcx> fmt::Display for TypeError<'tcx> {
                            found bound lifetime parameter {}", br)
             }
             Sorts(values) => ty::tls::with(|tcx| {
-                report_maybe_different(f, values.expected.sort_string(tcx),
-                                       values.found.sort_string(tcx))
+                write!(f, "expected {}", values.expected.sort_string(tcx))
```

Comment:
I know you just did all the work to change this, but this part of the change makes me nervous.  If there are other errors that will also need "expected __, found ___" changes for the label, maybe it makes sense to do all of those at once.

What does this change look like without changing this line?